apps/opendosm/pages/api/revalidate.ts

```diff
@@ -37,9 +47,45 @@ export default async function handler(
       .json({ error: "Revalidation failed", message: err.message, revalidated: [] });
   }
 }
+
+// Checks if route exists. Routes only valid for static pages
+const validate = (route: string): Promise<string> =>
+  new Promise((resolve, reject) => {
+    if (static_routes.includes(route)) resolve(route);
+    else reject(`Route does not exist or is not a static page. Route: ${route}`);
+  });
+
 // Rebuilds the relevant page(s).
 const rebuild = async (res: NextApiResponse, route: string, routes: string[]) =>
   new Promise(async (resolve, reject) => {
-    await res.revalidate(route).catch(e => reject(e));
-    resolve(true);
+    switch (route) {
+      case all_routes.POPULATION:
+      case all_routes.KAWASANKU:
+      case all_routes.HOUSEHOLD_INCOME_EXPENDITURE:
+      case `/ms-MY${all_routes.POPULATION}`:
+      case `/ms-MY${all_routes.KAWASANKU}`:
+      case `/ms-MY${all_routes.HOUSEHOLD_INCOME_EXPENDITURE}`:
+        await res.revalidate(route);
+        const result = revalidateWithStates(res, route);
+        routes.push.apply(routes, result);
+        resolve(true);
+        break;
+
```

Comment:
pls remove KAWASANKU from here. it doesnt share the common `[state]` pattern & also because it covers district, dun, parlimen level